metadata/content.go

```diff
@@ -103,10 +108,14 @@ func (cs *contentStore) Update(ctx context.Context, info content.Info, fieldpath
 	}
 	if err := update(ctx, cs.db, func(tx *bolt.Tx) error {
 		bkt := getBlobBucket(tx, ns, info.Digest)
-		if bkt == nil {
+		if bkt == nil && cs.db.SharedNamespace() != "" {
+			bkt = getBlobBucket(tx, cs.db.SharedNamespace(), info.Digest)
+			if bkt == nil {
+				return errors.Wrapf(errdefs.ErrNotFound, "content digest %v", info.Digest)
+			}
+		} else if bkt == nil {
 			return errors.Wrapf(errdefs.ErrNotFound, "content digest %v", info.Digest)
 		}
```

Comment:
This block of code seems to have repeated in 2 other places. Can we write a function to avoid the repetition?